src/lib/containers/FullTextSearch.tsx

```diff
@@ -0,0 +1,114 @@
+import { library } from '@fortawesome/fontawesome-svg-core'
+import { faSearch, faTimes } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Collapse } from '@material-ui/core'
+import React, { useEffect, useRef, useState } from 'react'
+import { TextMatchesQueryFilter } from '../utils/synapseTypes/Table/QueryFilter'
+import {
+  QueryWrapperChildProps,
+  QUERY_FILTERS_COLLAPSED_CSS,
+  QUERY_FILTERS_EXPANDED_CSS,
+} from './QueryWrapper'
+
+library.add(faSearch)
+library.add(faTimes)
+
+type InternalSearchProps = QueryWrapperChildProps
+
+// See https://sagebionetworks.jira.com/browse/PLFM-7011
+const MIN_SEARCH_QUERY_LENGTH = 3
+
+function FullTextSearch(props: InternalSearchProps) {
+  const [searchText, setSearchText] = useState('')
+  const searchInputRef = useRef<HTMLInputElement>(null)
+  const showSearchBar = props?.topLevelControlsState?.showSearchBar
+
+  useEffect(() => {
+    if (showSearchBar) {
+      searchInputRef.current?.focus()
+    }
+  }, [showSearchBar])
+
+  const search = (event: React.SyntheticEvent<HTMLFormElement>) => {
+    event.preventDefault()
+
+    // The HTML validation doesn't trim the string, so we add an extra check.
+    // We don't auto-trim the form text ourselves because the user may still be focused on the input.
+    if (searchText.trim().length < MIN_SEARCH_QUERY_LENGTH) {
+      searchInputRef.current?.setCustomValidity(
+        `Search term must have a minimum of ${MIN_SEARCH_QUERY_LENGTH} characters`,
+      )
```

Comment:
Is requiring a min length of a trimmed string good enough? I really don't want to worry about tokenizing the input.

Reply:
Yep, let's start with that!